packages/react-server-cli/src/compileClient.js

```diff
@@ -0,0 +1,222 @@
+import webpack from "webpack"
+import path from "path"
+import mkdirp from "mkdirp"
+import fs from "fs"
+import ExtractTextPlugin from "extract-text-webpack-plugin"
+
+// commented out to please eslint, but re-add if logging is needed in this file.
+//import {logging} from "react-server"
+//const logger = logging.getLogger(__LOGGER__);
+
+// compiles the routes file for browser clients using webpack.
+// returns a tuple of { compiler, serverRoutes }. compiler is a webpack compiler
+// that is ready to have run called, and serverRoutes is a path to the transpiled
+// server routes file, which can be required and passed in to
+// reactServer.middleware().
+// TODO: add options for sourcemaps.
+export default (routes,{
+	workingDir = "./__clientTemp",
+	routesDir = ".",
+	outputDir = workingDir + "/build",
+	outputUrl = "/static/",
+	hot = true,
+	minify = false,
+} = {}) => {
+	const workingDirAbsolute = path.resolve(process.cwd(), workingDir);
+	mkdirp.sync(workingDirAbsolute);
+	const outputDirAbsolute = path.resolve(process.cwd(), outputDir);
+	mkdirp.sync(outputDirAbsolute);
+
+	const routesDirAbsolute = path.resolve(process.cwd(), routesDir);
+
+	// for each route, let's create an entrypoint file that includes the page file and the routes file
+	let bootstrapFile = writeClientBootstrapFile(workingDirAbsolute);
+	const entrypointBase = hot ? [`webpack-dev-server/client?${outputUrl}`,"webpack/hot/only-dev-server"] : [];
+	let entrypoints = {};
+	for (let routeName of Object.keys(routes.routes)) {
+		let route = routes.routes[routeName];
+		let formats = normalizeRoutesPage(route.page);
+		for (let format of Object.keys(formats)) {
+			const absolutePathToPage = path.resolve(routesDirAbsolute, formats[format]);
```

Comment:
why do you use let for route and formats but const for absolutePathToPage?  Iiuc, you never change the reference for route or formats, so you could use const?  (I'm not really recommending a change, mostly just trying to catch up)